x-pack/legacy/plugins/uptime/public/components/monitor/ping_histogram/ping_histogram_container.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { PingHistogramComponent } from '../../common/charts';
+import { getPingHistogram } from '../../../state/actions';
+import { selectPingHistogram } from '../../../state/selectors';
+import { useGetUrlParams } from '../../../hooks';
+import { useMonitorId } from '../../../hooks';
+import { ResponsiveWrapperProps, withResponsiveWrapper } from '../../common/higher_order';
+
+interface Props {
+  height: string;
+}
+
+const Container: React.FC<Props & ResponsiveWrapperProps> = ({ height }) => {
+  const {
+    statusFilter,
+    absoluteDateRangeStart,
+    absoluteDateRangeEnd,
+    dateRangeStart: dateStart,
+    dateRangeEnd: dateEnd,
+  } = useGetUrlParams();
+
+  const dispatch = useDispatch();
+  const monitorId = useMonitorId();
+
+  const { loading, data, esKuery, lastRefresh } = useSelector(selectPingHistogram);
+
+  useEffect(() => {
```

Comment:
Thanks for migrating all these - I find the hooks usage much cleaner and easier to understand.